instrumentation/opentelemetry-instrumentation-threading/tests/test_threading.py

```diff
@@ -0,0 +1,122 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+from unittest import mock
+
+from packaging import version
+from opentelemetry import trace as trace_api
+from opentelemetry.instrumentation.threading import ThreadingInstrumentor
+from opentelemetry.test.test_base import TestBase
+from opentelemetry.trace import SpanKind, get_tracer
+
+
+class TestThreadingInstrumentor(TestBase):
+    def setUp(self):
+        super().setUp()
+        ThreadingInstrumentor().instrument()
+
+        self.tracer = get_tracer(__name__)
+
+    def tearDown(self):
+        super().tearDown()
+        # ThreadingInstrumentor().uninstrument()
+    def print_square(self, num):
+        with self.tracer.start_as_current_span("square"):
+            print("Square: {}" .format(num * num))
+
+    def print_cube(self, num):
+        with self.tracer.start_as_current_span("cube"):
+            print("Cube: {}" .format(num * num * num))
+
+    def print_square_with_thread(self, num):
+        with self.tracer.start_as_current_span("square"):
+            t2 = threading.Thread(target=self.print_cube, args=(10,))
+            print("Square: {}" .format(num * num))
+            t2.start()
+            t2.join()
+
+    def calculate(self, num):
+        with self.tracer.start_as_current_span("calculate"):
+            t1 = threading.Thread(target=self.print_square, args=(num,))
+            t2 = threading.Thread(target=self.print_cube, args=(num,))
+            t1.start()
+            t1.join()
+            
+            t2.start()
+            t2.join()
+
+
+    def test_without_thread_nesting(self):
+        t1 = threading.Thread(target=self.print_square, args=(10,))
+
+        with self.tracer.start_as_current_span("root"):
+            t1.start()
+            t1.join()
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+
+        target, root = spans[:2]
+        
+        self.assertIs(target.parent, root.get_span_context())
+        self.assertIsNone(root.parent)
+
+    def test_with_thread_nesting(self):
+        t1 = threading.Thread(target=self.print_square_with_thread, args=(10,))
+
+
+        with self.tracer.start_as_current_span("root"):
+            t1.start()
+            t1.join()
+
+        spans = self.memory_exporter.get_finished_spans()
+
+        self.assertEqual(len(spans), 3)
+
+        cube, square, root = spans[:3]
+        
+        self.assertIs(cube.parent, square.get_span_context())
+        self.assertIs(square.parent, root.get_span_context())
+        self.assertIsNone(root.parent)
+
+    def test_with_thread_multi_nesting(self):
```

Comment:
Some ASCII diagram for each test will make it easier for the reader to follow.

Reply:
I need some help here. How do I create ASCII diagram?

Reply:
Well, this was a suggestion to use your keyboard chars as I mentioned in the comments but this is not required. Please name your threading variable appropriately instead of `t1`. Otherwise PR is in pretty good shape.

Reply:
Done!